File: src/bg/util/leto/impl/LocaleStringId.java
package bg.util.leto.impl;

import java.util.Locale;
import java.util.Map;

public class LocaleStringId {
  private String mValue = null;
  private String mDefaultValue = null;
  public LocaleStringId(String id) {
      mValue = id;
  }
  public LocaleStringId(String id, String defaultValueIfNotFound) {
      mValue = id;
      mDefaultValue = defaultValueIfNotFound;
  }
  
  public LocaleStringId(String id, String defaultValueIfNotFound, Map<Locale, String> translations) {
      this(id, defaultValueIfNotFound);
      LocaleStrings.addTranslations(this, translations);
  }
  
  public String getId() {
      return mValue;
  }
  
  public String getDefaultValue() {
      return mDefaultValue;
  }
}

File: javascript/bg/util/leto/impl/LocaleStringId.js
if (require != null) {
 var LocaleStrings = require("./LocaleStrings.js");
}

function LocaleStringId(id, defaultValueIfNotFound, translations) {
  this.mValue = id;
  this.mDefaultValue = defaultValueIfNotFound;
  if (translations != null) { 
    LocaleStrings.addTranslations(this, translations);
  }

  
  this.getId = function () {
      return this.mValue;
  }
  
  this.getDefaultValue = function () {
      return this.mDefaultValue;
  }
}

if (module != null && module.exports != null)  {
  module.exports = LocaleStringId
}

